commit c2a3a1f03132c40f6eaf299986e980031827bcd3 (refs/changes/97/1097/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-12-08T15:38:44-08:00 (2 years, 10 months ago)
    
    joyent/node-cueball#45 want to see which nameservers returned which errors

diff --git a/lib/resolver.js b/lib/resolver.js
index 99aee31..f53c358 100644
--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -446,7 +446,7 @@ CueBallDNSResolver.prototype.state_srv_error = function (S) {
 			r.delay = r.maxDelay;
 
 	} else {
-		self.r_log.trace(self.r_lastError,
+		self.r_log.trace({ err: self.r_lastError },
 		    'repeated error during SRV resolution for service %s, ' +
 		    'will retry in 5min', self.r_service);
 
@@ -577,7 +577,7 @@ CueBallDNSResolver.prototype.state_aaaa_error = function (S) {
 			r.delay = r.maxDelay;
 
 	} else {
-		self.r_log.trace(self.r_lastError,
+		self.r_log.trace({ err: self.r_lastError },
 		    'repeated error during AAAA resolution for name %s, ' +
 		    'proceeding', self.r_srv.name);
 
@@ -679,7 +679,7 @@ CueBallDNSResolver.prototype.state_a_error = function (S) {
 			r.delay = r.maxDelay;
 
 	} else {
-		self.r_log.debug(self.r_lastError,
+		self.r_log.debug({ err: self.r_lastError },
 		    'repeated error during A resolution for name %s, ' +
 		    'proceeding', self.r_srv.name);
 
@@ -818,14 +818,15 @@ function srvKey(srv) {
 	return (hash.digest('base64'));
 }
 
-function NoNameError(name) {
+function NoNameError(cause, name) {
 	if (Error.captureStackTrace)
 		Error.captureStackTrace(this, NoNameError);
 	this.name = 'NoNameError';
-	this.message = 'No records returned for name ' + name;
 	this.dnsName = name;
+	mod_verror.VError.call(this, cause,
+	    'No records returned for name %s', name);
 }
-mod_util.inherits(NoNameError, Error);
+mod_util.inherits(NoNameError, mod_verror.VError);
 NoNameError.isInstance = function (i) {
 	return (i instanceof NoNameError);
 };
@@ -898,7 +899,7 @@ CueBallDNSResolver.prototype.resolve = function (domain, type, timeout) {
 			err.code = sorted[0];
 		}
 		if (err && err.code === 'NXDOMAIN')
-			err = new NoNameError(domain);
+			err = new NoNameError(err, domain);
 		if (!err && msg && msg.getAnswers().length === 0)
 			err = new NoRecordsError(domain, type);
 
